lib/ansible/module_utils/opennebula.py

```diff
@@ -73,15 +74,22 @@ def create_one_client(self):
         else:
             self.fail("Either api_url or the environment variable ONE_URL must be provided")
 
-        if self.module.params.get("api_username"):
+        if self.module.params.get("api_username") and self.module.params.get("api_password"):
             username = self.module.params.get("api_username")
-        else:
-            self.fail("Either api_username or the environment vairable ONE_USERNAME must be provided")
-
-        if self.module.params.get("api_password"):
             password = self.module.params.get("api_password")
         else:
-            self.fail("Either api_password or the environment vairable ONE_PASSWORD must be provided")
+            if self.module.params.get("api_auth_file"):
+                authfile = self.module.params.get("api_auth_file")
+            else:
+                authfile = path.join(environ.get("HOME"), ".one", "one_auth")
+            try:
```

Comment:
I really like the idea of checking for the `.one/one_auth`. What about to try also `http://localhost:2633` as a last resort default for api_url, like this:   https://github.com/xorel/ansible/commit/0d8c9599b6024205d3c40f384c9dd4a47823bace

Like so, it would make the setup seamless for most of the cases.

Reply:
you are correct this would mimic the standard behavior completely.
can i pull your suggestion so that your code is marked as yours ?

Reply:
sure, NP